### ISSUE TITLE
[Crypto] Implement Binance Bitcoin Exchange

### ISSUE THREAD

**Issue by AlexCatarino** [QuantConnect/Lean#2004]
Seeking a complete, tested [Binance](https://www.binance.com) bitcoin exchange implementation for QuantConnect, including the IDQH implementation for streaming binance data.  

1 - IBrokerage interface implementation for Binance. All IBrokerage tests should pass.
2 - IQueueDataHandler interface implementation for Binance streaming data.
3 - If possible IDataDownloader implementation for Lean/ToolBox for Binance API.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

**Comment by mysti9uemirage** [comment]
Do you happen to have an ETA for this? 
Will using https://github.com/glitch100/BinanceDotNet help the development process?

**Comment by fhassis** [comment]
Is there any progress on this work? I would like to try it if possible.

**Comment by omidkrad** [comment]
I found the branch for this feature is [here](https://github.com/Adalyat/Lean/tree/feature-2004/binance-brokerage/Brokerages/Binance) but looks like it is abandoned.

**Comment by Adalyat** [comment]
@fhassis basically it's done but some minor changes required. There is also a blocker required Lean core changes that should be merged before binance. I think we can afford that in mid/end of August.

**Comment by omidkrad** [comment]
That's great news @Adalyat. For Bunance support, are you using direct REST API calls or using any libraries?

Also I noticed there is this post [here](https://www.quantconnect.com/forum/discussion/4423/binance-crypto-trading-module) that a user says he has added Binance support to QuantConnect and operating. Maybe you can get a code review from him too.

**Comment by omidkrad** [comment]
I came across this project [ccxt-rest](https://ccxt-rest.io/) a few days ago. It's basically one REST interface to 135+ cryptocurrency exchanges. If we can add it as a broker to QuantConnect we can use all those exchanges.

**Comment by joao-neves95** [comment]
#3511 

There is a [CCXT C# .NET wrapper](https://github.com/lisa3907/ccxt.net) implementation, of the original [CCXT](https://github.com/ccxt/ccxt), that supports hundreds of exchanges.
It would be interesting to incorporate in Lean, as @omidkrad said.

**Comment by omidkrad** [comment]
@joao-neves95 I reviewed the CCXT C# project that you referenced and it seems to be a reimplementation of CCXT rather than a wrapper. It supports only 26 exchanges and doesn't seem to be maintained that well, so I wouldn't recommend it. That said, it is possible to create a C# wrapper over CCXT using [Edge.js](https://github.com/tjanczuk/edge) by running CCXT in Node and exposing its functions in-process to .NET, however, I prefer a microservice approach with [ccxt-rest](https://ccxt-rest.io/).

I've created a separate issue for that https://github.com/QuantConnect/Lean/issues/3793
